datafusion/src/physical_plan/functions.rs

```diff
@@ -234,6 +310,14 @@ impl BuiltinScalarFunction {
             BuiltinScalarFunction::Random | BuiltinScalarFunction::Now
         )
     }
+    /// Returns the [Volatility] of the builtin function.  
+    pub fn volatility(&self) -> Volatility {
+        match self {
+            BuiltinScalarFunction::Random => Volatility::Volatile,
+            BuiltinScalarFunction::Now => Volatility::Stable,
+            _ => Volatility::Immutable,
```

Comment:
perhaps it would be better to explicitly match each individual enum variant here so we don't accidentally set newly added variant to immutable?